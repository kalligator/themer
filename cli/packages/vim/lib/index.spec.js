const { render, renderInstructions } = require('./index'),
  { colors } = require('../../colors-default');

describe('themer vim color scheme generator', () => {
  const testColorSetConfiguration = (message, colors) => {
    it(message, async () => {
      const files = await Promise.all(render(colors, {}));
      const fileContents = files[0].contents.toString('utf8');
      expect(files.length).toBe(1);
      expect(/undefined/.test(fileContents)).toBe(false);
      expect(/'dark'/.test(fileContents)).toBe('dark' in colors);
      expect(/'light'/.test(fileContents)).toBe('light' in colors);
    });
  };

  testColorSetConfiguration(
    'should produce only one file containing both schemes if passed both a dark and light theme',
    colors,
  );
  testColorSetConfiguration(
    'should produce only one file containing only a dark scheme if passed only a dark color set',
    { dark: colors.dark },
  );
  testColorSetConfiguration(
    'should produce only one file containing only a light scheme if passed only a light color set',
    { light: colors.light },
  );

  it('should provide installation instructions', async () => {
    const files = await Promise.all(render(colors));
    const instructions = renderInstructions(files.map(({ name }) => name));
    expect(instructions).toMatchSnapshot();
  });
});
