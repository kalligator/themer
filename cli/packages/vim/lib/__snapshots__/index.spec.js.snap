// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themer vim color scheme generator should provide installation instructions 1`] = `
"
Copy or symlink \`ThemerVim.vim\` to \`~/.vim/colors/\`.

Then set the colorscheme in \`.vimrc\`:

    \\" The background option must be set before running this command.
    colo ThemerVim
"
`;
